components/camel-mongodb/src/main/java/org/apache/camel/component/mongodb/verifier/MongoComponentVerifierExtension.java

```diff
@@ -0,0 +1,101 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.mongodb.verifier;
+
+import java.util.Map;
+
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientOptions;
+import com.mongodb.MongoClientURI;
+import com.mongodb.MongoSecurityException;
+import com.mongodb.MongoTimeoutException;
+import org.apache.camel.component.extension.verifier.DefaultComponentVerifierExtension;
+import org.apache.camel.component.extension.verifier.ResultBuilder;
+import org.apache.camel.component.extension.verifier.ResultErrorBuilder;
+import org.apache.camel.component.extension.verifier.ResultErrorHelper;
+import org.apache.camel.component.mongodb.conf.MongoConfiguration;
+import org.bson.Document;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.camel.util.CastUtils.cast;
+
+public class MongoComponentVerifierExtension extends DefaultComponentVerifierExtension {
+    private static final Logger LOG = LoggerFactory.getLogger(MongoComponentVerifierExtension.class);
+
+    private static final int CONNECTION_TIMEOUT = 2000;
+
+    public MongoComponentVerifierExtension() {
+        super("mongodb");
+    }
+
+    @Override
+    public Result verifyParameters(Map<String, Object> parameters) {
+        ResultBuilder builder = ResultBuilder.withStatusAndScope(Result.Status.OK, Scope.PARAMETERS)
+                .error(ResultErrorHelper.requiresOption("host", parameters))
+                .error(ResultErrorHelper.requiresOption("user", parameters))
+                .error(ResultErrorHelper.requiresOption("password", parameters));
+        return builder.build();
+    }
+
+    @Override
+    public Result verifyConnectivity(Map<String, Object> parameters) {
+        return ResultBuilder.withStatusAndScope(Result.Status.OK, Scope.CONNECTIVITY)
+                .error(parameters, this::verifyCredentials)
+                .build();
+    }
+
+    private void verifyCredentials(ResultBuilder builder, Map<String, Object> parameters) {
+        MongoConfiguration mongoConf = new MongoConfiguration(cast(parameters));
+        MongoClientOptions.Builder optionsBuilder = MongoClientOptions.builder();
+        // Avoid retry and long timeout
+        optionsBuilder.connectTimeout(CONNECTION_TIMEOUT);
+        optionsBuilder.serverSelectionTimeout(CONNECTION_TIMEOUT);
+        optionsBuilder.maxWaitTime(CONNECTION_TIMEOUT);
+        MongoClientURI connectionURI = new MongoClientURI(mongoConf.getMongoClientURI(), optionsBuilder);
```

Comment:
Do you think it makes sense to check for `connectionURI` if is not null? My IDE analyser reports that `connectionURI.getDatabase()` might be null

Reply:
The `mongoConf` always provide a fallback database name if the user is not providing (ie `admin` db). I think the IDE complains because that method is marked on the driver as `@Nullable`.